libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotMessageHandler.cs

```diff
@@ -17,9 +18,20 @@ protected override async Task<InvokeResponse> ProcessMessageRequestAsync(HttpReq
         {
             var activity = default(Activity);
 
-            using (var bodyReader = new JsonTextReader(new StreamReader(request.Body, Encoding.UTF8)))
+            // Get the request body and deserialize to the Activity object.
+            // We need to leave the stream open here so others downstream can access it.
+            var originalPosition = request.Body.Position;
```

Comment:
I find the whole resetting of the position for the next person to be an anti-pattern. By definition, if multiple people are going to use the stream they need to make sure they position the Stream where they want/need it. Did you experience something that lead you to do this?

If not, I would recommend the recording of the `originalPosition` and the `try`/`finally`.